app/repositories/streams_repository.rb

```diff
@@ -3,11 +3,13 @@ def find(id)
     Stream.find(id)
   end
 
-  def calc_bounding_box!(stream)
-    stream.min_latitude = stream.measurements.minimum(:latitude)
-    stream.max_latitude = stream.measurements.maximum(:latitude)
-    stream.min_longitude = stream.measurements.minimum(:longitude)
-    stream.max_longitude = stream.measurements.maximum(:longitude)
+  def calc_bounding_box!(stream, calculate_bounding_box = Outliers::CalculateBoundingBox.new)
```

Comment:
Most of this stuff should be extracted to a service. But have no chance to make it happen now